packages/features/bookings/Booker/components/hooks/useBookings.ts

```diff
@@ -0,0 +1,279 @@
+import { updateQueryParam, getQueryParam } from "@calcom/features/bookings/Booker/utils/query-param";
+import {
+  createBooking,
+  createRecurringBooking,
+  mapBookingToMutationInput,
+  mapRecurringBookingToMutationInput,
+  createInstantBooking,
+  useTimePreferences,
+} from "@calcom/features/bookings/lib";
+import { getFullName } from "@calcom/features/form-builder/utils";
+import { useBookingSuccessRedirect } from "@calcom/lib/bookingSuccessRedirect";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { useRouterQuery } from "@calcom/lib/hooks/useRouterQuery";
+import { bookingMetadataSchema } from "@calcom/prisma/zod-utils";
+import { trpc } from "@calcom/trpc";
+import { showToast } from "@calcom/ui";
+
+import type { useBookingFormReturnType } from "./useBookingForm";
+import { useInitialFormValues } from "./useInitialFormValues";
+
+export interface IUseBookings {
+  event: useEventReturnType;
+  hashedLink?: string | null;
+  bookingForm: useBookingFormReturnType["bookingForm"];
+}
+
+export interface IUseBookingLoadingStates {
+  creatingBooking: boolean;
+  creatingRecurringBooking: boolean;
+  creatingInstantBooking: boolean;
+}
+
+export interface IUseBookingErrors {
+  hasDataErrors: boolean;
+  dataErrors: unknown;
+}
+
+export const useBookings = ({ event, hashedLink, bookingForm }: IUseBookings) => {
+  const router = useRouter();
+  const eventSlug = useBookerStore((state) => state.eventSlug);
+  const setFormValues = useBookerStore((state) => state.setFormValues);
+  const rescheduleUid = useBookerStore((state) => state.rescheduleUid);
+  const bookingData = useBookerStore((state) => state.bookingData);
+  const timeslot = useBookerStore((state) => state.selectedTimeslot);
+  const seatedEventData = useBookerStore((state) => state.seatedEventData);
+  const { t, i18n } = useLocale();
+  const bookingSuccessRedirect = useBookingSuccessRedirect();
+  const bookerFormErrorRef = useRef<HTMLDivElement>(null);
+  const [expiryTime, setExpiryTime] = useState<Date | undefined>();
+  const recurringEventCount = useBookerStore((state) => state.recurringEventCount);
+  const isInstantMeeting = useBookerStore((state) => state.isInstantMeeting);
+  const duration = useBookerStore((state) => state.selectedDuration);
+  const { timezone } = useTimePreferences();
+  const username = useBookerStore((state) => state.username);
+  const routerQuery = useRouterQuery();
+  const searchParams = useSearchParams();
+
+  const isRescheduling = !!rescheduleUid && !!bookingData;
+
+  const { initialValues, key } = useInitialFormValues({
+    eventType: event.data,
+    rescheduleUid,
+    isRescheduling,
+  });
+
+  const bookingId = parseInt(getQueryParam("bookingId") || "0");
+  const hasInstantMeetingTokenExpired = expiryTime && new Date(expiryTime) < new Date();
+  const _instantBooking = trpc.viewer.bookings.getInstantBookingLocation.useQuery(
```

Comment:
This constant is unused - should it be deleted or actually used?

Reply:
it's just there for it's on success effect I suppose